### DISCUSSION TITLE
How best to store the billing period of a Stripe Subscription

### DISCUSSION THREAD

**Discussion by mpvosseller** [pay-rails/pay#693] [General]
I'd like to store the "current billing period" of a Stripe Subscription so I can display a message like "Your plan renews on September 9, 2023.". Any recommendations on how best to do that?

The best I've come up with (at a high level) is:
1- Create a new model called `SubscriptionDetail` with a `belongs_to :subscription` relationship
2- Override the `Pay::Stripe::Subscription.sync` method by prepending something like:

```ruby
def sync(*args)
  pay_subscription = super
  update_subscription_detail(pay_subscription)
  pay_subscription
end

def update_subscription_detail(pay_subscription)
  pay_subscription.subscription_detail.current_period_start = pay_subscription.stripe_subscription[:current_period_start]
  pay_subscription.subscription_detail.current_period_end =  pay_subscription.stripe_subscription[:current_period_end]
  pay_subscription.subscription_detail.save!
end
 ```

Any better ideas or suggestions?

Thanks!

**Comment by excid3** [comment]
It would be nice to have `current_period_start` and `current_period_end` stored on the Subscription model. I'm not sure if we can reliably add these for Braintree or Paddle, but we can try.

Want to send a PR? I would probably add them as datetime columns.
